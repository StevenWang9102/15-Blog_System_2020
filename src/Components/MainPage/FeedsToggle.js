import React from "react";
// import { Link } from "react-router-dom";
// import { displayLimit, offset } from "../../Functions/HttpClient";
import { FeedsNav } from "../../Components/MainPage/FeedsNav";

export const FeedsToggle = props => {
let renderArray =[]
if(props.onPage === "User Profile") renderArray = ["My Articles", "Favorited Articles"];
if(props.onPage === "Article Preview") renderArray = ["Your Feed", "Global Feed", "Popular Feed"];

  return (
    <div className='articles-toggle'>
      <ul className='nav nav-pills outline-active'>
    {
        renderArray.map((feed, index) => {
            props.onPage === "User Profile"? ():()

        }
    }

        {/* {renderArray.map((feed, index) => {
          {props.onPage === "User Profile"? 
          (
            <li className='nav-item'>
              <FeedsNav
                navName={feed}
                author_name={props.author_name}
                setLoading={props.setLoading}
                setProfileNavStatus={props.setProfileNavStatus}
                profileNavStatusLeft={props.profileNavStatusLeft}
                profileNavStatusRight={props.profileNavStatusRight}
                loadUserProfileDetail={props.loadUserProfileDetail}
                onFavoritedArticleNavClicked={props.onFavoritedArticleNavClicked}
              />
            </li>
          ):(<li className='nav-item'>
          <FeedsNav
             navName={feed}
             userInformation={props.userInformation}
             setLoading={props.setLoading}
             loadYourFeedArticles={props.loadYourFeedArticles}
             setHomeNavStatus={props.setHomeNavStatus}
             yourNav={props.yourNav}
             loadGlobalFeeds={props.loadGlobalFeeds}
                globalNav={props.globalNav}
                popularNav={props.popularNav}
                currentTagName={props.currentTagName}
                onFavoritedArticleClicked={props.onFavoritedArticleClicked}/>
        </li>)            
          }         */}
          </ul>
    </div>
  );
};
