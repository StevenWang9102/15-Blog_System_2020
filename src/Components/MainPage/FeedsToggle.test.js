import React from "react";
import { FeedsToggle } from "./FeedsToggle";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <FeedsToggle
        fromPage='test'
        profileNavStatusLeft='test'
        profileNavStatusRight='test'
        loadYourFeedArticles={() => {}}
        setProfileNavStatus={() => {}}
      />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
