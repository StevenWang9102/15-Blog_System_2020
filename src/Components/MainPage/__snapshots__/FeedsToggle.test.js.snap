// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="articles-toggle"
>
  <ul
    className="nav nav-pills outline-active"
  />
</div>
`;
